File: src/main/java/fi/finlit/edith/ui/services/hibernate/PlaceDaoImpl.java
package fi.finlit.edith.ui.services.hibernate;

import static fi.finlit.edith.sql.domain.QPlace.place;

import java.util.Collection;

import fi.finlit.edith.sql.domain.Place;
import fi.finlit.edith.ui.services.PlaceDao;

public class PlaceDaoImpl extends AbstractDao<Place> implements PlaceDao {
    @Override
    public Collection<Place> findByStartOfName(String partial, int limit) {
        return query()
            .from(place)
            .where(place.normalized.last.startsWithIgnoreCase(partial)).limit(limit)
            .list(place);
    }

    @Override
    public void remove(Long placeId) {
        Place place = getById(placeId);
        getSession().delete(place);
    }

    @Override
    public void save(Place place) {
        getSession().save(place);
    }

    @Override
    public Place getById(Long id) {
        return query().from(place).where(place.id.eq(id)).uniqueResult(place);
    }

}

File: src/main/scala/fi/finlit/edith/ui/services/hibernate/PlaceDaoImpl.scala
package fi.finlit.edith.ui.services.hibernate

import fi.finlit.edith.sql.domain.QPlace.place
import java.util.Collection
import fi.finlit.edith.sql.domain.Place
import fi.finlit.edith.ui.services.PlaceDao
//remove if not needed
import scala.collection.JavaConversions._

class PlaceDaoImpl extends AbstractDao[Place] with PlaceDao {

  override def findByStartOfName(partial: String, limit: Int): Collection[Place] = {
    query.from(place).where(place.normalized.last.startsWithIgnoreCase(partial)).limit(limit).list(place)
  }

  override def remove(placeId: java.lang.Long) {
    var place = getById(placeId)
    getSession.delete(place)
  }

  override def save(place: Place) {
    getSession.save(place)
  }

  override def getById(id: java.lang.Long): Place = {
    query.from(place).where(place.id eq id).uniqueResult(place)
  }
}
